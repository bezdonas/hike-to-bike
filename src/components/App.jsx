import { h, Component } from 'preact';
import StationsMap from './StationsMap/StationsMap.jsx';
import Searchbar from './Searchbar/Searchbar.jsx';
import Sidebar from './Sidebar/Sidebar.jsx';
import Logo from './Logo/Logo.jsx';

const defaultPosition = [39.953, -75.171];
export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentPosition: defaultPosition,
    };

    this.handleChangePosition = this.handleChangePosition.bind(this);
  }
  handleChangePosition(newPosition) {
    this.setState({
      currentPosition: newPosition,
    });
  }
  render(props, state) {
    return (
      <div id="app">
        <Sidebar>
          <Logo />
          <Searchbar
            position={this.state.currentPosition}
            changePosition={this.handleChangePosition}
          />
        </Sidebar>
        <StationsMap currentPosition={this.state.currentPosition} />
      </div>
    );
  }
}
