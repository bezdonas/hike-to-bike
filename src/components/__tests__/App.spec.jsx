import { h, preact } from 'preact';
import render from 'preact-render-to-json';
import App from '../App.jsx';

describe('App smoke', () => {
  it('renders and matches snapshot', () => {
    expect(render(<App />)).toMatchSnapshot();
  });
});
