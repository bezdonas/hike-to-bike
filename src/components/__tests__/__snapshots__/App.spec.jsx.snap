// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App smoke renders and matches snapshot 1`] = `
<div
  id="app"
>
  <aside
    class="sidebar"
  >
    <h1
      class="logo-block"
    >
      Hike to aÂ bike
    </h1>
    <input
      class="searchbar"
      onChange={[Function]}
      type="text"
      value={
        Array [
          39.953,
          -75.171,
        ]
      }
    />
  </aside>
  <div
    class="stations-map"
    id="stations-map"
  />
</div>
`;
